actions: added nextjs-biomejs.yml for github actions

mertcanaltin opened the pull request:
I tried to make an actions for us to run and build the Next.js application in the github workflow with biomejs   #23

mertcanaltin committed:
Create nextjs-biomejs.js.yml

mertcanaltin committed:
added working-directory

anonrig reviewed (commented):
Tauri already has a github action https://github.com/tauri-apps/tauri-action. Can you use this and use pnpm?

Commit:
refactored action for tauri

mertcanaltin commented:
I tried to try, I hope I'm on the right track

anonrig commented on .github/workflows/nextjs-biomejs.js.yml:
```diff
@@ -0,0 +1,76 @@
+
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [14.x, 16.x, 18.x]
```
```suggestion

```

anonrig commented on .github/workflows/nextjs-biomejs.js.yml:
```diff
@@ -0,0 +1,76 @@
+        working-directory: ./client
+
+    - name: Install pnpm
+      run: npm install -g pnpm
```
Please use https://github.com/pnpm/action-setup

anonrig commented on .github/workflows/nextjs-biomejs.js.yml:
```diff
@@ -0,0 +1,76 @@
+      run: npm install -g pnpm
+
+    - name: Install Project Dependencies with pnpm
+      run: pnpm install
```
This should only run on client folder

mertcanaltin commented:
I wonder if there should be a command that clears the cache?

lexor commented on .github/workflows/nextjs-biomejs.js.yml:
```diff
@@ -0,0 +1,76 @@
+
+    - name: Run Biome.js
+      run: |
+         npm install -g @biomejs/biome
```
fyi: already installed `@biomejs/biome` (https://github.com/anonrig/calendar/blob/main/client/package.json#L36)

mertcanaltin committed:
Update nextjs-biomejs.js.yml

Co-authored-by: Yagiz Nizipli <yagiz@nizipli.com>

Commit:
chore: Update GitHub Actions workflow to use pnpm, remove global Biome.js installation

Commit:
chore: Update GitHub Actions workflow to use pnpm, remove global Biome.js installation

mertcanaltin commented:
Changes made in this commit:
- Replaced 'npm install -g pnpm' with 'pnpm/action-setup' for pnpm installation.
- Updated all relevant npm commands to use 'pnpm' within the 'client' directory.
- Removed the global 'npm install -g @biomejs/biome' installation.
- Directly used 'biome format' within the 'client' directory.
- Ensured that all dependencies are properly installed locally for both Next.js and Tauri within the 'client' directory.

anonrig commented on .github/workflows/nextjs-biomejs.js.yml:
```diff
@@ -0,0 +1,76 @@
+        version: 8
+
+    - name: Install Project Dependencies with pnpm in the client folder
+      run: pnpm install
```
You run this command twice

mertcanaltin replied:
thank you I removed it

anonrig commented on .github/workflows/nextjs-biomejs.js.yml:
```diff
@@ -0,0 +1,76 @@
+      working-directory: ./client
+
+    - name: Run Biome.js in the client folder
+      run: biome format
```
pnpm run lint

anonrig commented on .github/workflows/nextjs-biomejs.js.yml:
```diff
@@ -0,0 +1,76 @@
+      working-directory: ./client
+
+    - name: Test Tauri App in the client folder
+      run: pnpm run tauri dev
```
Is this correct? I don't think we should run dev mode in here.

mertcanaltin replied:
I updated this to pnpm run dev

Commit:
chore: Update GitHub Actions workflow

anonrig commented on .github/workflows/nextjs-biomejs.js.yml:
```diff
@@ -0,0 +1,72 @@
+      run: pnpm run tauri build
+      working-directory: ./client
+
+    - name: Test Tauri App in the client folder
+      run: pnpm run dev
+      working-directory: ./client
```
```suggestion
```

anonrig commented on .github/workflows/nextjs-biomejs.js.yml:
```diff
@@ -0,0 +1,72 @@
+jobs:
+  build-and-test:
+
+    runs-on: ubuntu-latest
```
```suggestion
    runs-on: macos-latest
```

anonrig commented on .github/workflows/nextjs-biomejs.js.yml:
```diff
@@ -0,0 +1,72 @@
+name: 'Build and Test Next.js with Biome.js, Tauri, and pnpm'
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
```
```suggestion
on:
  pull_request:
    types: [opened, synchronize]
    paths-ignore:
      - "**/*.md"
  push:
    branches:
      - main
    paths-ignore:
      - "**/*.md"
```

anonrig commented on .github/workflows/nextjs-biomejs.js.yml:
```diff
@@ -0,0 +1,72 @@
+
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v2
```
```suggestion
      uses: actions/checkout@v3
```

anonrig commented on .github/workflows/nextjs-biomejs.js.yml:
```diff
@@ -0,0 +1,72 @@
+    - name: Install pnpm using pnpm/action-setup
+      uses: pnpm/action-setup@v2
+      with:
+        version: 8
```
```suggestion
        version: latest
```

anonrig commented on .github/workflows/nextjs-biomejs.js.yml:
```diff
@@ -0,0 +1,72 @@
+      uses: actions/setup-node@v2
+      with:
+        node-version: ${{ matrix.node-version }}
+        working-directory: ./client
```
```suggestion
```

anonrig commented on .github/workflows/nextjs-biomejs.js.yml:
```diff
@@ -0,0 +1,72 @@
+      uses: actions/checkout@v2
+
+    - name: Setup Node.js
+      uses: actions/setup-node@v2
```
```suggestion
      uses: actions/setup-node@v3
```

anonrig commented on .github/workflows/nextjs-biomejs.js.yml:
```diff
@@ -0,0 +1,72 @@
+      run: pnpm run lint
+      working-directory: ./client
+
+    - name: Test Next.js App with pnpm in the client folder
+      run: pnpm test
+      working-directory: ./client
```
```suggestion

```

anonrig commented on .github/workflows/nextjs-biomejs.js.yml:
```diff
@@ -0,0 +1,72 @@
+      run: pnpm test
+      working-directory: ./client
+
+    - name: Set up Rust and Tauri
+      uses: actions/checkout@v2
+      with:
+        repository: tauri-apps/tauri
+        path: ./client
+
+    - name: Install Rust stable in the client folder
+      run: |
+        curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y --default-toolchain stable
+        source $HOME/.cargo/env
+      working-directory: ./client
```
```suggestion
      - name: Install Rust Toolchain
        uses: ./.github/actions/rustup
```

anonrig commented on .github/workflows/nextjs-biomejs.js.yml:
```diff
@@ -0,0 +1,72 @@
+        source $HOME/.cargo/env
+      working-directory: ./client
+
+    - name: Install Dependencies for Tauri (Ubuntu) in the client folder
+      if: matrix.node-version == 'ubuntu-latest'
+      run: |
+        sudo apt-get update
+        sudo apt-get install -y libgtk-3-dev libwebkit2gtk-4.0-dev libappindicator3-dev librsvg2-dev patchelf
+      working-directory: ./client
```
We don't build on ubuntu. We only run on macOS.

```suggestion

```

anonrig commented on .github/workflows/nextjs-biomejs.js.yml:
```diff
@@ -0,0 +1,72 @@
+        sudo apt-get install -y libgtk-3-dev libwebkit2gtk-4.0-dev libappindicator3-dev librsvg2-dev patchelf
+      working-directory: ./client
+
+    - name: Build Tauri App in the client folder
+      run: pnpm run tauri build
+      working-directory: ./client
```
```suggestion
    - name: Install tauri cli
      run: cargo binstall tauri-cli --version 2.0.0-alpha.11
    - name: Build
      run: cargo tauri build
```

mertcanaltin committed:
Update nextjs-biomejs.js.yml

Co-authored-by: Yagiz Nizipli <yagiz@nizipli.com>

mertcanaltin committed:
Update nextjs-biomejs.js.yml

Co-authored-by: Yagiz Nizipli <yagiz@nizipli.com>

mertcanaltin committed:
Update nextjs-biomejs.js.yml

Co-authored-by: Yagiz Nizipli <yagiz@nizipli.com>

mertcanaltin committed:
Update nextjs-biomejs.js.yml

Co-authored-by: Yagiz Nizipli <yagiz@nizipli.com>

mertcanaltin committed:
Update nextjs-biomejs.js.yml

Co-authored-by: Yagiz Nizipli <yagiz@nizipli.com>

mertcanaltin committed:
Update nextjs-biomejs.js.yml

Co-authored-by: Yagiz Nizipli <yagiz@nizipli.com>

mertcanaltin committed:
Update nextjs-biomejs.js.yml

Co-authored-by: Yagiz Nizipli <yagiz@nizipli.com>

mertcanaltin committed:
Update nextjs-biomejs.js.yml

Co-authored-by: Yagiz Nizipli <yagiz@nizipli.com>

mertcanaltin committed:
Update nextjs-biomejs.js.yml

Co-authored-by: Yagiz Nizipli <yagiz@nizipli.com>

mertcanaltin committed:
Update nextjs-biomejs.js.yml

Co-authored-by: Yagiz Nizipli <yagiz@nizipli.com>

mertcanaltin committed:
Update nextjs-biomejs.js.yml

Co-authored-by: Yagiz Nizipli <yagiz@nizipli.com>

mertcanaltin commented:
@anonrig I added your suggestions to the commit thank you very much

lexor commented on .github/workflows/nextjs-biomejs.js.yml:
Double extensions a bit confusing. Maybe we can rename like `nextjs-biomejs.yml`?

anonrig commented on .github/workflows/nextjs-biomejs.js.yml:
```diff
@@ -0,0 +1,54 @@
+      working-directory: ./client
+
+
+      - name: Install Rust Toolchain
```
the indentations and spacings are wrong.

Commit:
fix: rename file & lint

mertcanaltin commented:
I guess I couldn't install rust

```js
0s
Prepare all required actions
Getting action download info
Download action repository 'Swatinem/rust-cache@v2' (SHA:e207df5d269b42b69c8bc5[1](https://github.com/anonrig/calendar/actions/runs/6069445833/job/16515814317?pr=24#step:8:1)01da[2](https://github.com/anonrig/calendar/actions/runs/6069445833/job/16515814317?pr=24#step:8:2)6f7d[3](https://github.com/anonrig/calendar/actions/runs/6069445833/job/16515814317?pr=24#step:8:3)1feddb[4](https://github.com/anonrig/calendar/actions/runs/6069445833/job/16515814317?pr=24#step:8:4))
Run ./.github/actions/rustup
  with:
    clippy: false
    fmt: false
    docs: false
    restore-cache: true
    save-cache: false
    shared-key: warm
  env:
    PNPM_HOME: /Users/runner/setup-pnpm/node_modules/.bin
Run echo 'clippy: false'
  echo 'clippy: false'
  echo 'fmt: false'
  echo 'docs: false'
  echo 'restore-cache: true'
  echo 'save-cache: false'
  shell: /bin/bash --noprofile --norc -e -o pipefail {0}
  env:
    PNPM_HOME: /Users/runner/setup-pnpm/node_modules/.bin
clippy: false
fmt: false
docs: false
restore-cache: true
save-cache: false
Run mkdir client/out
mkdir: client/out: File exists
Error: Process completed with exit code 1.
```

lexor committed:
fix(ci): update nextjs biomejs jobs order

mertcanaltin commented:
thank you @lexor ❤️

mertcanaltin committed:
Update nextjs-biomejs.yml

lexor committed:
perf: install tauri-cli with binstall

mertcanaltin commented:
![enter image description here](https://media.tenor.com/yvb0o8CB-tcAAAAC/brooklyn99-brooklyn-nine-nine.gif)

lexor commented:
I updated workflow. Now `tauri-cli` installs w/ `cargo binstall`.

Before: https://github.com/anonrig/calendar/actions/runs/6087714052/job/16516874705 (20min~)
After: https://github.com/anonrig/calendar/actions/runs/6087993672/job/16517700531 (8.5min~)

mertcanaltin commented:
Thank you very much for contributing to my wish to withdraw

mertcanaltin commented:
thank you very much for helping my pr

mertcanaltin commented:
Greetings, I wonder if there is a need for this development @anonrig @lexor
